src/base/periodic_boundary_base.C

```diff
@@ -64,6 +70,44 @@ bool PeriodicBoundaryBase::is_my_variable(unsigned int var_num) const
 }
 
 
+
+bool PeriodicBoundaryBase::has_transformation_matrix() const
+{
+  return bool(_transformation_matrix);
+}
+
+
+
+const DenseMatrix<Real> & PeriodicBoundaryBase::get_transformation_matrix() const
+{
+  if(!has_transformation_matrix())
+  {
+    libmesh_error_msg("Transformation matrix is not defined");
+  }
+
+  return *_transformation_matrix;
+}
+
+
+
+void PeriodicBoundaryBase::set_transformation_matrix(const DenseMatrix<Real> & matrix)
+{
+  // Make a deep copy of matrix
+  this->_transformation_matrix = libmesh_make_unique<DenseMatrix<Real>>();
+  *(this->_transformation_matrix) = matrix;
```

Comment:
I was expecting that three lines to turn into one line, not two, but apparently that's our fault - we don't have a DenseMatrix copy constructor!

Presumably that was an intentional decision to prevent people from accidentally passing a copy instead of passing-by-reference, but still...

Reply:
Yes, I also expected there to be a copy constructor.